pkg/encoding/kzgEncoder/multiframe.go

```diff
@@ -0,0 +1,212 @@
+package kzgEncoder
+
+import (
+	"bytes"
+	"encoding/gob"
+	"errors"
+	"fmt"
+
+	rs "github.com/Layr-Labs/eigenda/pkg/encoding/encoder"
+	kzg "github.com/Layr-Labs/eigenda/pkg/kzg"
+	bls "github.com/Layr-Labs/eigenda/pkg/kzg/bn254"
+)
+
+// Sample is the basic unit for a verification
+// A blob may contain multiple Samples
+type Sample struct {
+	Commitment bls.G1Point
+	Proof      bls.G1Point
+	RowIndex   int // corresponds to a row in the verification matrix
+	Coeffs     []bls.Fr
+	X          uint // X is the evaluating index which corresponds to the leading coset
+}
+
+// generate a random value using Fiat Shamir transform
+// we can also pseudo randomness generated locally, but we have to ensure no adversary can manipulate it
+// Hashing everything takes about 1ms, so Fiat Shamir transform does not incur much cost
+func GenRandomness(samples []Sample) (bls.Fr, error) {
```

Comment:
`Randomness` -> `RandomFactor`? (it's the terminology used by the research post anyway)

Reply:
The prover can fool the verifier based on a randomness by modifying its input. However, since there is a deterministic relation between input and the randomness, the adversarial disperser cannot perform this attack